### ISSUE TITLE
Zend\Db\Adapter\Driver\*\Connection problems regarding transactions, commit, and rollback

### ISSUE THREAD

**Issue by Jakovitz** [zendframework/zendframework#6567]
There seems to be a discrepancy between the different DB drivers, on how they handle their `$this->inTransaction` state.

Pgsql seems to be the only driver that gets it right, and ends the transaction on both commit() and rollback() (by setting `$this->inTransaction = false`). In other words, I can't trust that code like

```php
if ($dbAdapter->getDriver()->getConnection()->inTransaction()) {
	$dbAdapter->getDriver()->getConnection()->rollBack();
}
```

will not throw exceptions, for not being in an transaction. I'm not sure about all drivers, but at least Mysqli and Pdo_mysql drivers should be fixed.

Also, it seems that the oci8 driver might have another issue, regarding this topic. Once the transaction have started, there's no way to end it, and have the driver use OCI_COMMIT_ON_SUCCESS again. I have no Oracle installation, so I haven't been able to verify this, though.

I'd do a pull request, but since the changes will not be backwards compatible, I'm not sure whether to issue it against master or develop (they're bugfixes, not feature changes, in my eyes).

**Comment by billortell** [comment]
This is resolved (finally) in 2.4.0rc1+ but still an issue that should've and could've been stopped by simply setting inTransaction=true after the oci_error() and oci_execute() calls were made - as done already in the Pdo Connection.php file (and others).  Ugh.

**Comment by GeeH** [comment]
This issue has been closed as part of the bug migration program as outlined here - http://framework.zend.com/blog/2016-04-11-issue-closures.html
